CorsixTH/Lua/dialogs/edit_room.lua

```diff
@@ -1105,15 +1105,19 @@ end
 --!param xpos (int) X position of the tile.
 --!param ypos (int) Y position of the tile.
 --!param player_id (int) Player id owning the hospital.
---!param flag_names (array) If set, array with four additional required properties.
+--!param flag_names (array) If set, array with arbitrary additional required properties.
 --!return Whether the tile is considered to be valid.
 local function validDoorTile(xpos, ypos, player_id, flag_names)
   local th = TheApp.map.th
 
   local tile_flags = th:getCellFlags(xpos, ypos)
   if not (tile_flags.buildable or tile_flags.passable or tile_flags.owner == player_id) then return false end
   if not flag_names then return true end
-  return tile_flags[flag_names[1]] and tile_flags[flag_names[2]] and tile_flags[flag_names[3]] and tile_flags[flag_names[4]]
+  local accumulate_flags = true
```

Comment:
You don't need this, if there is any false you can exit early: ```if not tile_flags[flagtype] return false end``` and ```return true``` at the end.